shortcode.php

```diff
@@ -330,3 +330,134 @@ function geoip_detect_shortcode_user_info($attr) {
     return geoip_detect2_shortcode_user_info_wpcf7('', 'geoip_detect2_user_info', true);
 }
 add_shortcode('geoip_detect2_user_info', 'geoip_detect_shortcode_user_info');
+
+/**
+ *
+ * Content Hiding
+ *
+ * Uses an enclosing shortcode (i.e. [shortcode]Content[/shortcode]). Shortcode attributes can be as follows:
+ * "timezone", "continent", "country", "region"/"state", "city", and "and_not".
+ *
+ * Each attribute may only appear once in a shortcode, and each attribute can only contain a single value, or
+ * the shortcode will break, showing the contents no matter what.
+ *
+ * Examples:
+ *
+ * `[geoip_detect2_hide_unless country="US" state="TX"]<h1>Title</h1>[/geoip_detect2_hide_unless]`
+ * 	    - OR -
+ * `[geoip_detect2_hide_unless country="US" region="TX"]<h1>Title</h1>[/geoip_detect2_hide_unless]`
+ * 	    - OR -
+ * `[geoip_detect2_hide_unless country="US" region="Texas"]<h1>Title</h1>[/geoip_detect2_hide_unless]`
+ * Only displays the inner content if the specified attributes match the visitor's country and region/state.
+ *
+ * `[geoip_detect2_hide_unless country="US" state="TX" and_not="Houston"]<h1>Title</h1>[/geoip_detect2_hide_unless]`
+ * Only displays the inner content if the specified attributes match the visitor's country and region/state,
+ * and the specified exclusion is not found in any possible GeoIP Detect field (e.g. visitor's country,
+ * city, timezone, etc.).
+ *
+ * `[geoip_detect2_hide_unless continent="North America"]<h1>Title</h1>[/geoip_detect2_hide_unless]`
+ * Only displays the inner content if the specified attributes match the visitor's continent.
+ *
+ */
+function geoip_detect2_shortcode_show_if($atts, $content = null, $shortcodeName = '') {
+	$expectedResult = ($shortcodeName == 'geoip_detect2_show_if') ? true : false;
+
+    $atts_array = shortcode_atts(array(
+        'timezone' => null,
+        'continent' => null,
+        'country' => null,
+        'most_specific_subdivision' => null,
+        'region' => null,
+        'state' => null,
+        'city' => null,
+        'and_not' => null,
+        'not_country' => null),
+        $atts);
+
+    $info = geoip_detect2_get_info_from_current_ip();
+    $criteria_test_flag = true;         // If set to false, nothing will display
+    $temp_attribute_values = [''];      // Temporarily stores each value of an attribute (if multiple)
```

Comment:
`[]` is correct, but only available in PHP 7. As I support down to PHP 5.4, we need a `array()` (without the new statement).